### ISSUE TITLE
Shrink main menu? And more..

### ISSUE THREAD

**Issue by rosepac** [sylhare/Type-on-Strap#218]
Hello, I have been changing the template of my page. I've had quite a few problems with the old yaml "tags" from the previous template. But it seems like I already made it work, [this is the repository](https://github.com/rosepac/ciberninjas).

I have fought and reviewed here, but I do not finish finding the key.  To be able to remove the pages from the menu or to be able to leave some selected pages.

There is a way to limit this, or how would be the most appropriate way to add the pages without appearing in the menu.

---

I have more problems. If you could help me, I would appreciate it. Because I would not like to spend much more time on configurations. I do not know, if it is better that there will be other problems or directly everything here, because it may be small things. For example:

- the search engine does not work.
- can the tag url be customized to make it work properly next to the tag links? Change the link, but obviously "internal links" redirect to tag / # - example (I'm still working to test if I can modify it).

---

Thank´s for all! 💖

**Comment by rosepac** [comment]
## Customizing the tag page URL. Rename the url of the tag page: Solved.

1. Change permalink yaml with the url you want. On the tags page `/pages/tags`
2. Change `"/tags"` in `/_includes/tags_list` in the code `<a class="button" href="{{ "/tags" | relative_url }}#{{ tag | cgi_escape }}">` by the name you want to give the page.

**Comment by sylhare** [comment]
Hi Thanks for using the theme. You have a very large amount of posts, so it's very valuable to hear about your experience :)

For the menu bar, if you want your pages to be [hidden](https://github.com/sylhare/Type-on-Strap#layout-page) and not appear there, you can set (If that's what you meant):

```yml
hide: true
```

Also if you want to have multiple pages, you may use the `portfolio` layout (you can change the name to "pages" or "other" for example) to display some of the page so that they would all be in one place and won't pollute your navigation bar.

From what I see, it seems you are using an older version of the theme, that may explain why you had issues with the tags and search engine Try pulling/forking the latest version and let me know how it goes.

**Comment by rosepac** [comment]
> Hi @rosepac
> 
> Thanks for using the theme. You have a very large amount of posts, so it's very valuable to hear about your experience :)
> 
> For the menu bar, if you want your pages to be [hidden](https://github.com/sylhare/Type-on-Strap#layout-page) and not appear there, you can set (If that's what you meant):
> 
> ```yaml
> hide: true
> ```
> 
> Also if you want to have multiple pages, you may use the `portfolio` layout (you can change the name to "pages" or "other" for example) to display some of the page so that they would all be in one place and won't pollute your navigation bar.
> 
> **edit**: Ah! from what I see you are missing `/assets/data/search.json` this file is necessary for the search to work.

Sorry, for my english.. 😿 And thanks for the help. The truth is that it had the theme of minimal mistakes, and with all the publications it started to load badly, despite being a great work as a template and I decided to look for something a little different.

I also have numerous collections implemented and it seems to me that I will have to adapt them to the "portfolio template" that is the idea, but little by little. I have already done a little test with my collection of ["wiki"](https://ciberninjas.com/wiki/)".

I suppose the idea of ​​creating a "portfolio page" as the basis for all other collections. It's a good idea, I keep it. Thank´s. 
 Although I would like to write posts again, I have already left the page abandoned for a while.

- Thanks for the "hide" on the pages, I don't know how I didn't notice before if I checked the readme file 300 times 🙈

---

- Over `search.json` .
I think the document, if you had it loaded. Similarly, I have rewritten and copied it directly from your template. 
you say you are not loading the .json file for some reason? Maybe the github path?
 Also modify the changes (change the title and url) on the search page.

I have been searching and discovered that it could be a bug in Cloudflare (a system called rockets). I've already tried disabling it and clearing the cache. and the search engine still doesn't work.

I also tried removing the Cloudflare "minimizer" for Javascript, and it still doesn't work.

I was thinking and reviewing now. The first different thing I have found between your theme and the one loaded by me. It's the Github path, mine loads on `/` yours on the path with the theme name `/ Type-on-strap` Could this have something to do with the javascript of the search engine? or the simple-jekyll-search plugin?

---

The truth is that this search engine, along with adsense, is placing the ads wherever they want.
 It has me a little lost. In the end they will spoil the little traffic I already had 😹

**Comment by sylhare** [comment]
From what I see:
- Your config having `/` as baseurl is correct and does not impact the search
- The js file for simple-jekyll-search is in assets/js/vendor and so it should not be impacted by cloudflare (because it's not hosted via cloudflare).

I'll try to run your site locally and see what can be the porblem

**Comment by sylhare** [comment]
Ok so I ran your website locally, I found a couple of errors that you might need to check:

             Error: could not read file _posts/Noticias_Actualidad/2019-05-15-👩🎓-▷-cursos-de-udemy-❤️-cupones-gratis-✅-2019-en-inglés-parte-1.md: No such file or directory @ rb_sysopen - _posts/Noticias_Actualidad/2019-05-15-👩🎓-▷-cursos-de-udemy-❤️-cupones-gratis-✅-2019-en-inglés-parte-1.md
             Error: could not read file _posts/Noticias_Actualidad/2019-05-18-👩🎓-▷-cursos-de-udemy-🔥-cupones-gratis-✅-2019-en-inglés-parte-2.md: No such file or directory @ rb_sysopen - _posts/Noticias_Actualidad/2019-05-18-👩🎓-▷-cursos-de-udemy-🔥-cupones-gratis-✅-2019-en-inglés-parte-2.md
             Error: could not read file _posts/Libros/2019-10-09-🤾♂️-mentalidad-mamba-de-kobe-bryant.md: No such file or directory @ rb_sysopen - _posts/Libros/2019-10-09-🤾♂️-mentalidad-mamba-de-kobe-bryant.md
             Error: could not read file _posts/Artículos-Traducidos/2018-03-20-👩💻-aprender-a-programar-es-cada-vez-más-difícil.md: No such file or directory @ rb_sysopen - _posts/Artículos-Traducidos/2018-03-20-👩💻-aprender-a-programar-es-cada-vez-más-difícil.md
             Error: could not read file _posts/Artículos-Traducidos/2019-10-24-🕵️♀️-más-opciones-para-ayudar-a-los-sitios-web-a-obtener-una-vista-previa-de-su-contenido-en-la-búsqueda-de-google.md: No such file or directory @ rb_sysopen - _posts/Artículos-Traducidos/2019-10-24-🕵️♀️-más-opciones-para-ayudar-a-los-sitios-web-a-obtener-una-vista-previa-de-su-contenido-en-la-búsqueda-de-google.md
             Error: could not read file _posts/2020-02-24-👩💻-habitos-basicos-desarrollador-software.md: No such file or directory @ rb_sysopen - _posts/2020-02-24-👩💻-habitos-basicos-desarrollador-software.md

Maybe it's a work in progress while you wanted to move some stuff around. It means you may link to those post in some page while those files are not there.

I found the issue for your search page, I will add a fix to it.
Basically it was because of this article:

```yaml
---
title: >
    Flexciton utiliza la inteligencia artificial para ayudar a las fábricas a optimizar las líneas de producción
published: true
```

The `>` makes it escape the json and break the search. You can change it to in the mean time, that will fix the search:

```yml
---
title: Flexciton utiliza la inteligencia artificial para ayudar a las fábricas a optimizar las líneas de producción
```

And now it works fine locally:

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/20642750/87324271-de166100-c4fd-11ea-9047-5f1aa11b2d18.png">

**Comment by sylhare** [comment]
Should be fixed now 😄

**Comment by rosepac** [comment]
> Should be fixed now 😄

Sorry for being late to answer. During these days. I had already corrected several pages with bad yaml tags. I spent 3 or 4 days reviewing the yaml tags. And I couldn't find the ugly mistake.

You don't know what I love you 💖, a billion thanks 😻💖😻💖😻

---

Look, I've already graced some of the pages in the collections. [guides](https://ciberninjas.com/guias/), [wiki](https://ciberninjas.com/wiki/)

---

A silly doubt, using thumbnail + feature-img. Should generate image thumbnail for social media?
 I've been passing some links from your blog. To try..

It is rare because when trying to share on Facebook, sometimes the link box appears but the image is black. 
![image](https://user-images.githubusercontent.com/4309005/88810515-bd186600-d1b5-11ea-97ef-ba9aae412aa9.png)

And with other links. Load "a box" that says github error..
![image](https://user-images.githubusercontent.com/4309005/88810596-dc16f800-d1b5-11ea-9b2b-9562b3d0107e.png)

Some idea for this? 🙈
